eng/pipelines/upstream-sync-pipeline.yml

```diff
@@ -59,5 +59,6 @@ extends:
                     -github-user microsoft-golang-bot `
                     -github-pat $(BotAccount-microsoft-golang-bot-PAT) `
                     -github-pat-reviewer $(BotAccount-microsoft-golang-review-bot-PAT) `
+                    -auto-merge false `
```

Comment:
We should be sure this works before hard-coding it and potentially blocking ongoing auto-sync. It should work fine to push this change internally and run the pipeline on the dev branch to see what happens.

Reply:
I've triggered https://dev.azure.com/dnceng/internal/_build/results?buildId=2567448&view=logs&s=ddda3e96-b7f2-5f09-e4e5-8b5fbdbd8459 to test